pkg/resolver/client/ens/impl.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ens
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/ethclient"
+	goens "github.com/wealdtech/go-ens/v3"
+)
+
+func wrapDial(ep string) (*ethclient.Client, error) {
+
+	// Open a connection to the ethereum node through the endpoint.
+	cl, err := ethclient.Dial(ep)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure the ENS resolver contract is deployed on the network we are now
+	// connected to.
+	if _, err := goens.PublicResolverAddress(cl); err != nil {
+		return nil, err
+	}
+
+	return cl, nil
+}
+
+func wrapResolve(backend bind.ContractBackend, name string) (string, error) {
+
+	// Connect to the ENS resolver for the provided name.
+	ensR, err := goens.NewResolver(backend, name)
```

Comment:
why create this on  each resolve call? is  it not  better just  to reuse  it?

Reply:
'NewResolver' will return an ENS resolver which is specific for a given domain name. Internally, it perform checks and fetches the contract address so that it can be queried for ENS records.

Caching is another matter that IMHO should be discussed separately.